cmd/tempo-cli/cmd-query-blocks-summary.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+
+	"context"
+	"fmt"
+
+
+
+
+	"github.com/grafana/tempo/pkg/model/trace"
+
+	//"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/util"
+	// "github.com/grafana/tempo/tempodb"
+	// "github.com/grafana/tempo/tempodb/backend"
+	// "github.com/grafana/tempo/tempodb/encoding"
+	// "github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type queryBlocksSummaryCmd struct {
+	backendOptions
+
+	TraceID  string `arg:"" help:"trace ID to retrieve"`
+	TenantID string `arg:"" help:"tenant ID to search"`
+}
+
+func (cmd *queryBlocksSummaryCmd) Run(ctx *globalOptions) error {
+	r, _, c, err := loadBackend(&cmd.backendOptions, ctx)
+	if err != nil {
+		return err
+	}
+
+	id, err := util.HexStringToTraceID(cmd.TraceID)
+	if err != nil {
+		return err
+	}
+
+	results, err := queryBucket(context.Background(), r, c, cmd.TenantID, id)
+	if err != nil {
+		return err
+	}
+
+	var (
+		combiner   = trace.NewCombiner()
+		// marshaller = new(jsonpb.Marshaler)
+		// jsonBytes  = bytes.Buffer{}
+	)
+
+	fmt.Println()
+	for i, result := range results {
+		combiner.ConsumeWithFinal(result.trace, i == len(results)-1)
+	}
+
+	combinedTrace, _ := combiner.Result()
+	size := 0
+	spanCount := 0
+	firstStartTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].StartTimeUnixNano
+	lastEndTime := combinedTrace.Batches[0].ScopeSpans[0].Spans[0].EndTimeUnixNano
+	rootSpan := combinedTrace.Batches[0].ScopeSpans[0].Spans[0]
+	rootSpanResource := combinedTrace.Batches[0].Resource
+	rootServiceName := ""
+	serviceNameMap := make(map[string]int)
+	for _, attr := range rootSpanResource.Attributes {
+		if "service.name" == attr.Key {
+			rootServiceName = attr.Value.GetStringValue()
+		}
+	}
+
+	for _, b := range combinedTrace.Batches {
+		size += b.Size()
+		for _, attr := range b.Resource.Attributes {
+			if "service.name" == attr.Key {
+				serviceNameMap[attr.Value.GetStringValue()] ++
+			}
+		}
+		for _, scope := range b.ScopeSpans {
+			spanCount += len(scope.Spans)
+			for _, span := range scope.Spans {
+				if span.EndTimeUnixNano > lastEndTime {
+					lastEndTime = span.EndTimeUnixNano
+				}
+			}
+		}
+	}
+
+
+	// get top 5 most frequent service names
+	lowest := 0
+	topFiveName := make([]string, 5)
```

Comment:
i would probably just put all of the key/value pairs in a slice and sort it. it won't be as efficient, but it will be clearer and this CLI command doesn't need to be super efficient.